### PR TITLE
Skip controller params that are part of the route

### PR DESCRIPTION
**Pull request by frederikbosch**
We have the following route.
```php
$this->controller(['/directdebit/{companyId:i}/{yearId:i}', 'directdebit'], DirectDebitController::class);
```

But the following fails due to Cannot use the same placeholder companyId twice.

```php
public function getTransferDelete ($companyId, $yearId, $id) {
```

The companyId and yearId parameter should be skipped when generating a route for a controller because they are part of the route itself.

### CODE REVIEW

**Comment by frederikbosch** [comment]
If you like this proposal, I can add tests for them.

**Comment by mrjgreen** [comment]
I think you may just be using this wrong. The `controller` method maps all "RESTful" type methods to routes automatically.

I'm fairly sure you should just be using:

~~~PHP
$this->controller('/', DirectDebitController::class);
~~~

**Comment by frederikbosch** [comment]
Hmm, maybe. Consider the following. I have resource called company, which has multiple year resources and the year resource has multiple directdebit resources. This generates a tree.

The controller method only allows there to be one resource. What I want is child resources. A solution with the current code would be (like you are proposing).

```php
$this->controller('/direct-debit', DirectDebitController::class);
```
Then the routes look like this.
/direct-debit/{companyId:i}/{yearId:i}/{directDebitId:i}

But the following is a better representation of the resources.
/company/{companyId:i}/year/{yearId:i}/direct-debit/{directDebitId:i}

This would require the controller route to support the following.

```php
$this->controller('/company/{companyId:i}/year/{yearId:i}/direct-debit/{directDebitId:i}', DirectDebitController::class);
```

Basically, it adds the possibility to use subroutes.

**Comment by mrjgreen** [comment]
Okay, I like the idea of this. I'm not sure about the implementation as it stands, but I will try and work this in a later date.

**Comment by frederikbosch** [comment]
Sound reasonable!

**Comment by frederikbosch** [comment]
Implementation should be better then this PR. I added a feature request for this. Examples of Ember are added.
